docs/version-locking.md

```diff
@@ -0,0 +1,129 @@
+# Helix Pages Version and Feature Locking
+
+This document describes the version locking feature for helix pages. 
+
+The goal of this feature is to provide customers a way to select, test, and develop against a selected _version_ or _feature_ of helix pages.
+
+There are several aspects to this feature:
+
+- Definition of a version set (by the helix developer team)
+- Rendering of pages using a specific version set
+- Selecting default version set for a helix pages project
+- Developer experience of a helix pages customer
+
+## Definition of a Version Set
+
+The parts that influence how a resource is rendered in helix pages are:
+
+- a set of serverless services (dispatch, content-proxy, static, redirect, word2md, gdocs2md, etc)
+- the helix pages code (pre.js)
+- the helix-pipeline version
+- the VCL deployed via helix-publish to fastly (inner-cdn config)
+- the project code (helix-query.yaml, head.html, scripts, css, etc...)
+
+A helix pages _version_ or _feature_ is defined by:
+
+- an identifier (eg. breaking-december)
+- a set of services with specific versions (eg. word2md v20201201.1, content-proxy v20201201.4)
+- a helix pages code branch
+
+The version set definition is recorded in a `versions.yaml` which is stored in the respective helix pages code branch. For example, a `breaking-december` _version_ would have `versions.yaml` in the `breaking-december` branch in the helix-pages repo:
```

Comment:
Should we call it `helix-versions.yaml` to align with other Helix config files?

Reply:
right now, I think its only something used in the helix-pages project and not used by customers... but yes...maybe.